paddle/memory/memory.cc

```diff
@@ -77,6 +85,36 @@ size_t Used<platform::GPUPlace>(platform::GPUPlace place) {
   return GetGPUBuddyAllocator(place.device)->Used();
 }
 
+template <>
+void Copy<platform::CPUPlace, platform::GPUPlace>(platform::CPUPlace, void* dst,
+                                                  platform::GPUPlace,
+                                                  const void* src, size_t num,
+                                                  cudaStream_t stream) {
+  platform::GpuMemcpyAsync(dst, src, num, cudaMemcpyDeviceToHost, stream);
+}
+
+template <>
+void Copy<platform::GPUPlace, platform::CPUPlace>(platform::GPUPlace, void* dst,
+                                                  platform::CPUPlace,
+                                                  const void* src, size_t num,
+                                                  cudaStream_t stream) {
+  platform::GpuMemcpyAsync(dst, src, num, cudaMemcpyHostToDevice, stream);
+}
```

Comment:
Here, use `cudaMemcpyDeviceToHost` and `cudaMemcpyHostToDevice`, `cudaMemcpyDeviceToDevice`. But the cudaMemcpyKind  of `cudaMemcpyDefault`:

> cudaMemcpyDefault = 4
> Direction of the transfer is inferred from the pointer values. Requires unified virtual addressing

Can we use `cudaMemcpyDefault` to simplify code？